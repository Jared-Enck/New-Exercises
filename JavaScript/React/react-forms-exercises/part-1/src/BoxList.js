import React, {useState} from 'react'
import { v4 as uuid } from 'uuid'
import Box from './Box'
import NewBoxForm from './NewBoxForm'

function BoxList() {
    const [boxes, setBoxes] = useState([])
    
    const addBox = (box) => {
        const newBox = {...box, id: uuid()}
        setBoxes(boxes => [...boxes, newBox])
    }
    return (
        <>
            <NewBoxForm addBox={addBox} />

            <div className='row mx-auto'>
                <div className='row mx-auto grid gap-4'>
                    {boxes.map(box => 
                        <Box 
                            key={uuid()}
                            color={box.color} 
                            width={box.width} 
                            height={box.height} 
                        />
                    )}
                </div>
            </div>
        </>
    )
}

export default BoxList