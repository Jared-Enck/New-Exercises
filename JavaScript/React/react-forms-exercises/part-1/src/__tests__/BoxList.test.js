import React from "react";
import { render } from "@testing-library/react";
import BoxList from "../BoxList";
import BOXES from "../boxes"

it("renders without crashing", function() {
  render(<BoxList boxes={BOXES} />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<BoxList boxes={BOXES} />);
  expect(asFragment()).toMatchSnapshot();
});