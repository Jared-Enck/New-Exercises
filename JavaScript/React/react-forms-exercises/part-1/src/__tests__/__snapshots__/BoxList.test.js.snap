// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row mx-auto"
  >
    <div
      class="col-6"
    >
      <form
        class="form-control mb-3 shadow-sm bg-light"
      >
        <label
          for="color"
        >
          Color
        </label>
        <input
          class="form-control mb-2"
          id="color"
          name="color"
          placeholder="Enter a Color"
          type="text"
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          class="form-control mb-2"
          name="width"
          placeholder="Enter width ex: 100px"
          type="text"
        />
        <label
          for="height"
        >
          Height
        </label>
        <input
          class="form-control mb-2"
          name="height"
          placeholder="Enter height ex: 100px"
          type="text"
        />
        <button
          class="btn btn-medium btn-primary"
        >
          Add Box!
        </button>
      </form>
    </div>
  </div>
  <div
    class="row mx-auto"
  >
    <div
      class="row mx-auto grid gap-4"
    />
  </div>
</DocumentFragment>
`;
